Commands/GenerateWebpImages.php

```diff
@@ -2,22 +2,93 @@
 
 namespace FroshWebP\Commands;
 
+use FroshWebP\Components\ImageStack\Arguments;
+use FroshWebP\Components\WebpEncoderInterface;
+use FroshWebP\Factories\WebpConvertFactory;
+use FroshWebP\Models\WebPMedia;
+use FroshWebP\Repositories\WebPMediaRepository;
 use FroshWebP\Services\WebpEncoderFactory;
+use Shopware\Bundle\MediaBundle\MediaService;
 use Shopware\Commands\ShopwareCommand;
+use Shopware\Components\Model\ModelManager;
 use Symfony\Component\Console\Helper\ProgressBar;
 use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
 use Symfony\Component\Console\Output\OutputInterface;
 
 /**
  * Class GenerateWebpImages
  */
 class GenerateWebpImages extends ShopwareCommand
 {
+    /**
+     * @var ModelManager
+     */
+    private $modelManager;
+
+    /**
+     * @var WebPMediaRepository
+     */
+    private $webpRepository;
+
+    /** @var WebpEncoderFactory */
+    private $encoderFactory;
+
+    /** @var WebpEncoderInterface[] */
+    private $runnableEncoders;
+
+    /** @var int */
+    private $webpQuality;
+
+    /** @var MediaService */
+    private $mediaService;
+
+    /**
+     * GenerateWebpImages constructor.
+     *
+     * @param ModelManager $manager
+     * @param MediaService $mediaService
+     * @param WebpEncoderFactory $webpEncoder
+     * @param $webpConfig
+     */
+    public function __construct(ModelManager $manager, MediaService $mediaService, WebpEncoderFactory $webpEncoder, $webpConfig)
+    {
+        $this->modelManager = $manager;
+        $this->mediaService = $mediaService;
+        $this->webpRepository = $manager->getRepository(WebPMedia::class);
+        $this->encoderFactory = $webpEncoder;
+        $this->runnableEncoders = WebpEncoderFactory::onlyRunnable($this->encoderFactory->getEncoders());
+        $this->webpQuality = $webpConfig['webPQuality'];
+        parent::__construct();
+    }
+
     protected function configure()
     {
         $this
             ->setName('frosh:webp:generate')
-            ->setDescription('Generate webp images for all orginal images');
+            ->setDescription('Generate webp images for all orginal images. Can also run as stack-execution,
+            for specific folders only and with excluded folders')
+            ->addOption('stack', 's', InputOption::VALUE_OPTIONAL, 'process amount per iteration')
+            ->addOption('offset', 'o', InputOption::VALUE_OPTIONAL, 'process amount per iteration')
+            ->addOption('force', 'f', InputOption::VALUE_NONE, 'forces recreation')
+            ->addOption('setCollection', 'c', InputOption::VALUE_OPTIONAL | InputOption::VALUE_IS_ARRAY,
```

Comment:
why not using the common Shopware wording 'albumid'?

Reply:
I didn't use it because on my opinion it would occur to much.
But if it is desired, I can change it